%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% Welcome to Overleaf --- just edit your article on the left,
% and we'll compile it for you on the right. If you give
% someone the link to this page, they can edit at the same
% time. See the help menu above for more info. Enjoy!
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% For more detailed article preparation guidelines, please see:
% http://f1000research.com/author-guidelines

\documentclass[10pt,a4paper,twocolumn]{article}
\usepackage{f1000_styles}

%% Default: numerical citations
\usepackage[numbers]{natbib}

%% Uncomment this lines for superscript citations instead
% \usepackage[super]{natbib}

%% Uncomment these lines for author-year citations instead
% \usepackage[round]{natbib}
% \let\cite\citep

\usepackage{subcaption}
\usepackage{hyperref}
\usepackage{pdfpages}
\usepackage[normalem]{ulem}
\newcommand{\DONE}[1]{\begingroup\color{blue}#1\endgroup}
\newcommand{\FA}[1]{\begingroup\color{magenta}#1\endgroup}
\newcommand{\ILH}[1]{\begingroup\color{blue}#1\endgroup}
\newcommand{\TODO}[1]{\begingroup\color{red}#1\endgroup}
%opening
\begin{document}
	
	
	\title{\textit{MSF: Modulated Sub-graph Finder} }
	
	\author[1]{Mariam R. Farman}
	\author[1]{Ivo L. Hofacker}
	\author[1,2]{Fabian Amman}
	\affil[1]{Institute for Theoretical Chemistry,Theoretical Biochemistry Group,University of Vienna, Austria}
	\affil[2]{Department of Chromosome Biology, Max F. Perutz Laboratories,University of Vienna, Austria}
	
	
	
	\maketitle
	\thispagestyle{fancy}
	
	\begin{abstract}
		
		High throughput techniques such as RNA-seq or microarray analysis have
		proven to be invaluable for the characterization of global transcriptional
		gene activity changes due to external stimuli or diseases. Differential
		gene expression analysis (DGEA) is the first step in the course of data
		interpretation, typically producing lists of dozens to thousands of
		differentially expressed genes. To further guide the interpretation of
		these lists, different pathway analysis approaches have been
		developed. These tools typically rely on the classification of genes into
		sets of genes, such as pathways, based on the interactions between the genes
		and their function in a common biological process. Regardless of technical
		differences, these methods do not properly account for cross talk between
		different pathways and rely on binary separation into differentially
		expressed gene and unaffected genes based on an arbitrarily set \textit{p}-value
		cut-off.
		
		To overcome this limitation, we developed a novel approach to identify
		concertedly modulated sub-graphs in the global cell signaling network,
		based on the DGEA results of all genes tested. Thereby, expression patterns
		of genes are integrated according to the topology of their interactions and
		allow potentially to read the flow of information and identify the effectors. The described software, named \texttt{Modulated
			Sub-graph Finder} (\texttt{MSF}) is freely available at
		\url{https://github.com/Modulated-Subgraph-Finder/MSF}.
		
	\end{abstract}
	
	\section*{Keywords}
	
	Differential gene expression analysis; pathway analysis; combining \textit{p}-value; cell signaling network;
	
	\clearpage
	
	\section*{Introduction}
	
	High throughput sequencing techniques have been widely used to yield
	differentially expressed genes (DEG)~\cite{DEG}. The changes in
	transcript abundance are measured, e.g.~by next generation sequencing
	techniques and interpreted as an indicator of differential expression of
	genes. DEGs can be used to gain insights into the mechanisms underlying differences between conditions of samples, such as healthy versus infection. Differential gene expression analysis (DGEA) informs about the
	magnitude of expression changes, which are often
	expressed as log-fold change. The sign of log-fold change and the confidence level of
	observing an authentic change, often expressed as \textit{p}-value. The information from DEGs is further interpreted to extract meaningful biological
	insights. For example, genes that could be involved in the response to a
	particular stimulus or may be the cause of an infection. To this end, pathway-based analysis has become an important tool to further interpret
	the results of a DGEA and to acquire understandings of the perturbations in
	a biological system. These pathway-based methods use predefined pathways or
	networks which are sets of genes with their
	interactions forming a functional unit. DEGs help to identify pathways or
	networks that may be altered during an infection providing
	important information about diseases and its treatment
	process~\cite{Khatri2012}. The
	expression measurements of the genes obtained from DGEA in combination with
	statistical methods and the predefined pathways are used to identify specifically modulated
	pathways and processes~\cite{Campos}. 
	
	
	Well established resources for pathway annotations are KEGG (Kyoto
	Encyclopedia of Genes and Genomes)~\cite{Kegg} and Reactome~\cite{Reactome}. KEGG pathways is a branch of KEGG database that
	hosts a collection of manually drawn pathway maps representing the
	molecular interaction, reaction and relation networks of cellular
	functions. Similarly, Reactome is an open-source, manually curated,
	peer-reviewed database for signaling and metabolic molecules with their
	interactions formed into different biological pathways~\cite{Reactome}. Both provide predefined pathways which are sets of
	genes and their interactions categorized into functional units.
	
	
	Existing pathway-based analysis approaches use different research
	designs, which can be categorized into ORA (Over-representation
	analysis), FCS (Functional class scoring) and pathway topology based
	methods. All of which aim to find a subset of genes, e.g.,
	significantly differentially expressed genes, genes associated with a
	certain pathway more often than expected given the total set of
	examined genes, e.g.~the whole genome background. ORA is
	the first and the most basic method of pathway analysis~\cite{Campos}. It uses a
	DEG list with user defined cut-off for the log-fold change and
	\textit{p}-value (most commonly using absolute log-fold change $\geq$
	2 and \textit{p}-value $\leq$ 0.05). Subsequently, sets of genes
	associated with annotated pathways are tested for being
	over-represented in the set of DEGs. To this end, hyper-geometric
	distribution, chi-square tests, binomial probability or the Fisher’s
	exact test are used. Thereby the information of the topology of genes
	in the pathways is neglected~\cite{Bayer}. Furthermore, ORA assumes
	that the biological pathways are independent of each other and ignores
	the fact that they cross-talk and overlap~\cite{Khatri2012,Campos}.
	
	Unlike ORA, FCS has no artificial cut-off to define a DEG list. FCS works in
	three steps. First it calculates the gene-level statistics including
	correlation of molecular measurements using differential expression of
	individual genes, ANOVA, t-test and Z-score. In the second step the
	statistics of individual genes in a pathway are transformed to an
	individual pathway-level statistic commonly using Kolmogorov-Smirnov
	statistic, mean or median. Finally the statistical significance of the
	pathway-level statistics is assessed. Although FCS covers some of the
	limitations from ORA, it still ignores the topology of genes in a pathway,
	cross-talk and overlap of the pathways~\cite{Khatri2012,Campos}. Pathway
	topology based methods are similar to FCS except that they consider the
	topology of each gene during the gene-level statistics but still lacks to aim
	to link the different functional pathways~\cite{Khatri2012}.
	
	\DONE{Recently systems biology approaches are also helping illuminate the structural and functional design of a cell~\cite{mitra_carvunis_ramesh_ideker_2013}. Mitra and colleagues discuss the different systems biology approaches to identify and measure the modules of inter-molecular interactions steering a particular biological process. These approaches are classified into four categories, identification of "active modules" with molecular profiles, identification of "conserved modules" across species , identification of "differential modules" with multiple conditions and identification of "composite modules" with interactions of different types. Identification of the active modules is one of the most successful approaches, which overlays the modules or networks onto the molecular profiles (transcriptomics, genomics, proteomics) to identify the "active modules". To find these active modules heuristics solutions like simulated annealing (SA), greedy methods, genetic algorithms (GA), network propagation and co-clustering methods are used~\cite{mitra_carvunis_ramesh_ideker_2013}. JActiveModules~\cite{jActiveModules} has been the first of this kind using simulated annealing to find the active modules.}
	
	\DONE{The motivation was to find the modulated biological processes cross-talking, for which identifying the modulated networks first with the help of systems biology was necessary.} On these grounds we propose a novel approach to make use of the rich gene
	and protein interaction annotation resources available, to gain additional
	functional insights from basic DGEA. To this, we start with the
	presupposition that expression of neighboring genes within a functional
	pathway are not independent from each other. Rather, they are often
	regulating each others expression or are part of the same
	regulon~\cite{Michalak}. We understand that the categorization of links
	between genes into labeled pathways is often an arbitrary one, given the
	extensive cross talk between different pathways. Although these categories
	have proven to be useful in many situations, they force a certain
	perspective onto the interpretation of novel data. Based on these two
	principles, we aim to find sub-graphs of connected genes within cell
	signaling network, which exhibit as a whole significant differential
	expression changes. This approach differs in two main aspecarg1ts from common
	pathway analysis tools. First, it does not aim to identify functional
	pathways enriched in differentially expressed genes, but detects sub-graphs
	or branches in a network graph (potentially spanning more than one
	functionally grouped pathway) which is coherently modulated. Second, it
	aims to improve the DGEA on the gene level, by collecting the information
	of neighboring genes, which as a whole might exhibit prominent enough
	signal to be called; again as a whole, significantly modulated. All of this can be helpful to understand the cause and effect of a stimulus and might inform about potential points of intervention.
	
	As input, information on functional links between genes provided by
	e.g.~KEGG or Reactome and information on the differential expression status
	of single genes resulting from a DGEA, are required. As a result the
	analysis returns sub-graphs and their joint confidence scores, reflecting
	how the perturbation is migrated through the network. Furtharg1ermore, the
	entry points of perturbation in the networks and overlap with conventional
	pathway categories are returned. These entry points are given impact scores showing the degree of perturbations introduced by them. The reliability of the entry points is checked using a T-test by investigating the significant difference between the downstream and upstream genes of each entry point. The output is prepared in a directed
	adjacency file, convenient for visualization, e.g., with
	StringApp~\cite{StringApp}, available as a Cytoscape plug-in~\cite{Cyto}.
	
	The proposed algorithm is named
	\texttt{Modulated Sub-graph Finder} (abbreviated
	\texttt{MSF}). \texttt{MSF} can help transform the information obtained
	from DGEA into comprehensible knowledge of signal transduction of genes and
	thereby being a valuable complement to existing pathway based
	methods.  \texttt{MSF} is freely accessible on github under the terms of the
	Creative Commons Attribution 4.0 International License.
	
	
	
	
	\section*{Methods}
	
	\texttt{MSF} was implemented as a java program. It is developed as a novel heuristic approach to find concertedly
	modulated sub-graphs in networks of biological interactions.  \texttt{MSF}
	does not use predefined gene sets grouped into functional units, but rather
	relies purely on the network of interacting genes. The input network
	consists of nodes corresponding to genes and edges representing
	interactions. Furthermore it utilizes comprehensive results from a
	differential gene expression analysis to discover the sub-graphs, or
	modules, which are as a whole modulated.
	
	\texttt{MSF} uses the individual gene's \textit{p}-values generated from
	the DGEA. The \textit{p}-value expresses the probability that the null
	hypothesis of unmodified gene expression can't be rejected for a given
	statistical model. To find significantly modulated sub-graphs individual
	\textit{p}-values of the vicinal genes in the global network are combined
	into a single combined \textit{p}-value, using a statistical method for
	combining dependent \textit{p}-values described by
	Hartung~\cite{Hartung}. Hartung's method uses the inverse of standard
	normal distribution function. Using the inverse normal cumulative distribution function $\Phi^{-1}$, individual gene \textit{p}-values $T_{i}$ are 
	transformed to their corresponding normal score $t_{i}=\Phi^{-1}(1-T_{i})$ that is uniformly distributed on (0,1). Then using these normal
	scores, the correlation between genes is calculated $Cov(t_{i},t_{j})=\rho$. The normal scores and
	correlation are applied to the weighted inverse normal function to calculate the
	combined \textit{p}-value $t(\rho)$ for all genes examined, namely the examined
	sub-graph
	\newline
	\begin{center}
		$t(\rho)=\frac{\sum_{i=1}^{n}\lambda_i t_{i} }{\sqrt{(1-\rho) \sum_{i=1}^{n} \lambda^{2}+\rho(\sum_{i=1}^{n} \lambda_i)^{2}}}$
	\end{center}
	
	Lambda $\lambda$ be the weights for each gene, currently all genes have equal weight i.e. 1. The combined \textit{p}-value  $t(\rho)$ of a sub-graph will express the
	significance of all genes in the sub-graph being modulated
	together. Thereby, the information from the different genes are used as,
	although not independent, replicated measurement of the behavior of the
	whole sub-graph. This potentially increases the power to detect also
	significant sub-graphs consisting of genes which are not significant on
	there own.
	\newline
	
	\subsection*{Overview of our method}
	
	To reduce the complexity to score all possible connected sub-graphs
	\texttt{MSF} applies a four step heuristic as described in the
	following. The proceeding identification of modulated sub-graphs from a
	network by \texttt{MSF} is presented as a flowchart diagram
	(Fig.~\ref{fig:pseudocodemsf}). \newline
	
	
	\textbf{Initializing modulated sub-graphs}
	
	\texttt{MSF} constructs the first sub-graph starting with the genes
	associated with the lowest (most significant) \textit{p}-value deduced from
	the DGEA. From this seed it tries to extend the sub-graph by adding directly
	neighboring genes, starting with the next most significant one. A single
	combined \textit{p}-value is calculated for the extended sub-graph.
	If the combined \textit{p}-value is smaller than the \textit{p}-value of
	the original sub-graph,
	the extended sub-graph is accepted. This step is
	iteratively repeated until no further extension is accepted. In this case
	the process starts over with all remaining genes not yet in the significantly
	modulated sub-graph. This step identifies all simple sub-graphs that
	are modulated in the whole network.\newline
	
	\textbf{Extending modulated sub-graphs}
	
	In the next step, we check if any of the initial modulated
        sub-graphs could further be extended beyond the immediate
          neighbouring genes. Instead of testing single genes and
          their compatibility to be added, groups of genes are
          considered. If the combined \textit{p}-value of the
        initial modulated sub-graph and the extension genes is smaller
        than the \textit{p}-value of the initial sub-graph the
        extension is accepted. All possible extension paths up to
        \emph{3} (default 2) genes at all nodes in the sub-graph are
        tested. Again, this step is iteratively repeated until no
        further genes are added to the significant differentially
        expressed sub-graphs. This step bridges small gaps of genes
        without a clear differential signal in the DGEA.\newline
	
	\textbf{Merging modulated sub-graphs}

        After detection and extension of the modulated sub-graphs,
          each pair of so far identified sub-graphs is tested if its
          combination scores better than each on its own. The merging of
        the sub-graphs is done by depth first search traversal from
        one sub-graph to the other sub-graph. If the two sub-graphs
        merge with the connector of at most \emph{3} genes (default 2
        gene) and the combined \textit{p}-value of the merged
        sub-graph including the bridging genes in between is less than
        the individual \textit{p}-values of the two sub-graphs, the
        two sub-graphs are merged together to one bigger
        modulated sub-graph. This step is repeated iteratively until
        no sub-graphs can be merged
          anymore.\newline
	
	\textbf{Finding sources \& sinks}
	
	In a last post processing step \texttt{MSF} identifies the
        trigger points of the modulated sub-graphs. These trigger
        genes are the sources of the sub-graphs with only outgoing
        edges. These genes can be interpreted as the possible entry
        points of perturbation from where the stimulus causes
        downstream effects. Each individual source is given an impact
        score, expressing the relative number of downstream genes
          within the corresponding sub-graph directly connected by
          directed links. This score can be interpreted as an upper
          limit of how much of the sub-graph's perturbation could have
          been introduced by the respective source, and thereby could
          be helpful to prioritize different identified sources for
          larger sub-graphs. In the same
        spirit as sources, the most downstream genes of the
        modulated sub-graphs are identified and defined as sinks. Due
        to loops not all sub-graphs are guaranteed to have sources or
        sinks. The reliability of each source is inspected using a T-test. The significant difference between the two groups, genes downstream the source to the genes upstream of the source is determined. This would help to assess if the source identified is reliable and marks the border between two different regulation regimes.
	
	
	\textbf{MSF output}
	
	\texttt{MSF} generates a directed network file as an output,
        containing complete directed interactions of all modulated
        sub-graphs identified. This file could be imported into
        Cytoscape~\cite{Cyto} for visualization. Additionally, a
          file containing details on all sources and sinks for all
          modulated sub-graphs is reported. Furthermore, for
          facilitated visualisation in Cytoscape, a node attribute
          file is provided, containing the source weightage and the
          log-fold chances of all considered genes.
	
	\textbf{Operation}
	
	\DONE{The only system requirements to run MSF are Java version 8 and JDK 1.8. The few package
		dependencies are already been added to the release. To run
		MSF, the user must provide two text files, one containing the
		DGEA from edgeR~\cite{edgeR} or DESeq2~\cite{love2014moderated}} and  the second one containing
        directed interactions in an adjacency format file. Example
        files and a detailed tutorial to use MSF has been provided on
        github https://github. com/Modulated-Subgraph-Finder/MSF.
	
	
	\section*{Results}
	
	\subsection*{Case study}
	
	To demonstrate its usefulness, \texttt{MSF} is applied to
        RNA-seq data set of primary human monocyte-derived macrophages
        (MDMs) infected with Ebola virus
        (GSE84188)~\cite{Olejnik}. Ebola Virus (EBOV) belongs to the
        Filoviridea family: filamentous, enveloped and single stranded
        RNA viruses. EBOV causes hemorrhagic fever in humans, inducing
        the host innate and adaptive immune response to be unable to
        control virus infection~\cite{Prins}. Currently, there are no
        approved antiviral drugs for the treatment of Ebola virus
        infection~\cite{Konde,Rhein}.  The initial targets of EBOV are
        the macrophages and dendritic immune
        cells~\cite{Falasca,Rhein}. EBOV inhibits the critical innate
        immune response of the host, which includes the activation of
        alpha/beta interferon (IFN-$\alpha /
        \beta$)~\cite{Prins,Konde,Cardenas}. It has been proposed that
        IFN-$\alpha / \beta$ should be tested against Ebola for its
        antiviral activity through clinical trials~\cite{Konde}. Ebola
        infection data was selected to test the approach because it
        has been well recognized for the last several decades, and
        vast literature is available for the pathogenesis of Ebola,
        thereby, facilitating the verification of the results of
        \texttt{MSF} with the vast literature present on Ebola
        infection. Especially, the detection of IFN-$\alpha / \beta$
        as point of action for the virus, could be considered as a
        basic indicator of the correctness and usefulness of the
        approach.
	
	EBOV infection count data was downloaded from GEO (GSE84188),
        it describes the course of infection at three time-points 6,
        24 and 48 hour post infection (hpi). Differential gene
        expression analysis was performed on the count data with edgeR
        package (version 3.4.2)~\cite{edgeR} using an upper-quartile
        normalization. The DEG analysis results generated by edgeR
        were used as input for \texttt{MSF}. Cell signaling
        interactions were filtered from Reactome Functional
        interactions (FIs) Version 2016~\cite{Cytokegg} for only
        direct interactions, which was used as a second input for
        \texttt{MSF}.
	
	For the earliest time point at 6~hpi, three large modulated
        sub-graphs were identified with 42, 139, and 69 genes.  The
        modulated sub-graphs consist predominantly of cytokines,
        chemokines (CXCL10, CCL8, CXCL9, CXCL11, CXCR4, CCR7, CCL4L1,
        CCL3L1, CCL4, CCL8, CCL20, CCL3, CCL19) and Interleukin genes
        (IL6, IL27, IL23).  IFNB1 and IFNA1 were both identified as
        two of the possible sources in the most significantly
        modulated sub-graph identified with 42 genes. The impact score
        of IFNB1 is 14.5\%, \DONE{the highest impact score and IFNA1 is 8.7\%, in top 5 highest impact scores in the sub-graph they
        	belong to} . Most of the sources identified by \texttt{MSF}
        were type~I interferon induced genes (Supplement material).
        At 24~hpi seven modulated sub-graphs were identified with four
        main sub-graphs consisting of 61, 222, 130 and 242 genes,
        others being smaller than 6 genes. Again, IFNB1 and IFNA1 were
        identified as two sources out of the total sources with 3.9\% 
        and 1.6\% impact score. \DONE{IFNB1 was one of the top 5 impact score sources for the corresponding sub-graph}. For the last time-point 48~hpi, six
        modulated sub-graphs were identified. Three of the sub-graphs
        were less than ten genes and main sub-graphs had 217, 224 and
        276 genes. IFNB1 and IFNA1 were identified as sources in the
        most significantly modulated sub-graph with an impact score of
        2.8\% and 3.7\%, \DONE{but not the highest ranked sources} (Supplement material). 
	
	As stated earlier IFN-$\alpha / \beta$ was reported to be one
        of the target genes of Ebola infection. We were able to
        successfully identify IFNA1 and IFNB1 as sources in all three
        Ebola infection time-points. Although IFNA1 and IFNB1 were
        already among the most significant genes in the DGEA during
        the later time points, \texttt{MSF} was also able to detect
        them as a source in the very early time-point when the genes
        were not significant based on the individual DGEA
        alone. Identifying the possible sources will reduce the search
        space for potential target genes and can help the biologist as
        the starting point of clinical testing for drugs and vaccines
        against an infection.
	
	Table~\ref{tab:rawVsHartung} compares the results of
        \texttt{MSF}, namely the number of detected sub-modules and
        their total genes numbers, to a simple analysis of mapping
        significantly modulated genes from the DGEA to the network and
        joining neighbors to modules. The numbers indicate that
        \texttt{MSF} detects less but larger and interpretable
        sub-modules, applying its statistical test. Furthermore, the
        dependency of the results from the \textit{p}-value cutoff
        choice is demonstrated for the DGEA, which is avoided for
        \texttt{MSF} altogether. It showcases how applying different
        cut-offs to the \textit{p}-value of genes from edgeR to the
        sub-graphs identified by \texttt{MSF} breaks the larger
        sub-graphs to many smaller unconnected sub-graphs, many of
        which are single genes.
	
	\subsection*{Modulated sub-graphs at 6~hpi}
	
	Three main modulated sub-graphs identified by \texttt{MSF} at
        6~hpi are shown in Fig.~\ref{fig:Sub-graph6hpi}. The graphs
        represent the immediate output of the \texttt{MSF}-analysis,
        visualized by StringApp~\cite{StringApp} in
        Cytoscape~\cite{Cyto}. Each node represents a gene part of a
        modulated sub-graph, whereby the associated colors code the
        functional annotation deduced from KEGG Pathways. The
        cross-talk between the pathways and also the multiplhttps://github.com/Modulated-Subgraph-Finder/MSF/tree/master/bine
        employment of many genes is evident. The flow of information
        between the sensors and effectors can be perceived given the
        directionality of each interaction, indicated by arrows.  In
        more detail, sub-graph~1 (bottom) shows how the activation of
        toll-like receptor, cytokine, chemokine activating jak-stat
        and mapk genes, together with TNF leads into apoptosis. The
        next significant sub-graph (sub-graph~2: top right) reveals
        how information from the Extra-cellular matrix (ECM) receptor,
        which are reported to interact with Ebola glycoprotein
        (GP)~\cite{Veljkovic}, chemokines, cytokines, and cytosolic
        DNA sensing are directly or indirectly controlling cell
        growth, differentiation, proliferation and apoptosis. It
        suggests that dysregulation of these pathways is responsible
        for modulation of apoptosis. Eventually, sub-graph~3 (top
        left) demonstrates how INFA1 and INFB1 modulates onchttps://github.com/Modulated-Subgraph-Finder/MSF/tree/master/bine more,
        via only a few intermediate steps, the apoptotic response of
        the cell. On the other side cAMP signaling genes activates
        platelet genes.
	
	This display case might advertise with how little effort
        complex data can be processed and prepared for
          interpretation by the domain expert, to apprehend the dynamics of the underlying processes
        and suggest testable hypothesis and potential points of
        intervention.
	
	\subsection*{Robustness}
	
	A potential concern is how noise in the gene expression
        measurements affects our analysis. To assess the robustness
        and stability of our method, we therefore added Poisson
        distributed noise to the read counts of the three time-points
        data set, used above.  Then DGEA was carried out on the
        disturbed data with the same parameters as for the native data
        using edgeR, followed by analysis with \texttt{MSF}. This
        procedure was carried out 100 times.  Every time the genes
        from the modulated sub-graphs identified from noisy data were
        compared to the genes of sub-graphs identified from the native
        data. The robustness of \texttt{MSF} analysis for the
        time-point 6, 24, and 48~hpi is shown in
        Fig.~\ref{fig:DEGvsMSF}. The procedure how data noise was
        modeled can be considered as rather stringent as \texttt{MSF}
        is sensitive to \textit{p}-value changes across the whole
        range of possible values. The observed median recall rates lay
        between 71~\% (6~hpi) and 84~\% (48~hpi).
	
	\subsection*{Benchmark}
	
	 \DONE{The purpose of benchmark is mainly to show overall capabilities of \texttt{MSF}. \texttt{MSF} was compared to jActivemodules~\cite{jActiveModules} since they use similar approach to find and score the modules. Comparison of \texttt{MSF} to Reactome~\cite{Reactome} was done because  Reactome is the standard tool used when looking for modulated pathways after DGEA. Since \texttt{MSF} and GSEA~\cite{Subramanian15545} use no cut-off to the DGEA, their results were also compared.}
	
	\textbf{jActiveModules}
	
	jActiveModules~\cite{jActiveModules} is a plugin in Cytoscape that searches for molecular
        interaction network to find expression activated
        sub-networks. The method used to score the expression
        activated sub-networks is close to the method used in
        \texttt{MSF}.  \DONE{The difference is in how these sub-networks are identified. \texttt{MSF} starts building the sub-graphs from one gene, incorporating and combining the \textit{p}-value of the next gene, with the check that the combined \textit{p}-value of new sub-graph should be better than the original. On the other hand jActiveModules first transforms all the gene’s \textit{p}-values to z-scores and tries to find connected sets of genes with unexpectedly high levels of differential expression, in this case high z-scores. The overall score of the sub-network is calculated by combining the z-scores of the genes. Next using their extended simulated annealing method jActiveModules finds the user defined number of highest scoring putative modules by toggling multiple nodes to merge additional components.} 
	
	The first time-point of Ebola infection data was analyzed
        using jActiveModules (Version 3.2.1) to compare the modulated
        sub-graphs identified by \texttt{MSF} and jActiveModules. The
        input files were same for both tools, number of modules
        parameter was set to 5 (default). This would report 5 putative modules even when only one of them is good. Out of the 5 putative modules identified by jActiveModules, the module with the highest pathScore was selected for comparison with
        \texttt{MSF} identified modulated sub-graphs. The module consists of a single graph with 314
        genes. While \texttt{MSF} identified three directed modulated
        sub-graphs with 42, 69 and 139 genes. The overlap of the
        common genes identified between \texttt{MSF} and jActiveModules
        is shown in Fig~\ref{fig:MSFvsJActiveModule}. \DONE{JActiveModules identified a single module, since this module had unexpectedly high levels of differential expression. With the extended simulated annealing method, jActiveModules toggled multiple nodes to merge a number of components with low score into this single high-scoring component making it larger than original.} Although jActiveModules was able to identify
        single connected module, \texttt{MSF} provides directionality,
        with the identification of possible perturbation sources of
        the sub-graphs. 
        
     

	\textbf{Reactome pathway analysis}
	
	Gene enrichment analysis was performed using Reactome analyze
        data tool~\cite{Reactome} (version 67) on the different time-points of Ebola infection data. Reactome's over-representation analysis tool tests
        whether certain Reactome pathways are enriched for the list of
        genes submitted to it. Genes from \texttt{MSF} identified
        sub-graphs for each time-point were analyzed for gene
        enrichment using this tool.  For comparison the DEG results
        from edgeR for the three time-points were filtered using the
        cut-off of adjusted \textit{p}-value < 0.05. This
        DEG list was used for gene enrichment analysis. The
        compression of \texttt{MSF} identified sub-graphs gene list
        and the DEG list analysis is shown in
        Fig~\ref{fig:msfvsreactome}
	
	All enriched pathways with a cut-off of \textit{p}-value
        < 0.05 for \texttt{MSF} and DEG list for the
        three time-points were selected. The comparison shows most of
        the pathways known from literature to be dis-regulated by
        Ebola infection are enriched in both the enrichment
        analysis. Toll-Like receptor signaling pathway when interacts
        with EBOV glycoprotein (GP), it triggers the activation of
        cytokines~\cite{Olejnik}. Toll-like receptor pathway is
        expected to be dis-regulated in the early stage of infection,
        this pathway was not identified as significantly dis-regulated
        when \textit{p}-value cut off DEG list was analyzed for
        enrichment. Nine toll-like receptor cascades TLR10, TLR2,
        TLR3, TLR4, TLR5, TLR7/8, TLR9, TLR1:TLR2 and TLR6:TLR2 were
        identified as dis-regulated from gene enrichment analysis of
        \texttt{MSF} identified sub-graph genes, not a single one of
        these cascade was shown to be dis-regulated in pathway
        enrichment analysis from DEG cut-off list. Since \texttt{MSF}
        considers the complete DEG results, even the weak signal at
        the earliest time-point was detected; for example Toll-like
        receptor signaling. While \texttt{MSF} is able to catch weak
        signals, it does not provide information about the functional
        relationships among genes like Reactome tool.
	
	\textbf{Gene set enrichment analysis (GSEA)}
	
	Gene set enrichment analysis (GSEA)~\cite{Subramanian15545}
        is a method to identify
        classes of genes or proteins that are over-represented in a
        large set of genes or proteins. GSEA uses statistical
        approaches to identify significantly enriched or depleted
        groups of genes. The complete DEG list from DGEA of the first
        time-point 6 hpi was analyzed using bioconductor package
        GSEABase (version 1.44.0). GSEA was able to identify
        Toll\FA{-}like recpetor, chemokine signaling pathway,
        cytosolic DNA-sensing pathway, Jak-STAT signaling pathway,arg1
        RIG-I-like receptor signaling pathway and apoptosis as the
        highest ranked pathways. Although GSEA identified the
        important pathways for Ebola infection, it did not show how
        the cross-talk and topology of genes between the pathways
        identifed. \texttt{MSF} and GSEA uses complete DEG list
        without any cut-off, is why pathways important for Ebola
        infection showed up even with weak signal genes in it.
	
	\section*{Discussion}
	
	Classic pathway analysis tools aim to detect in lists of
        significantly deregulated genes enriched associations with
        pathway genes categorized by their biological function and
        their interactions. Depending on the tool, the internal
        pathway topology is considered or neglected all together. The
        here presented tool, \texttt{MSF}, employs a different
        approach, by aiming to detect sub-graphs in whole gene
        regulatory networks which are significantly deregulated in a
        concerted manner. To this end, neighboring genes in the user
        provided network are tested for jointly common
        regulation. Exploiting that each gene's abundance, although
        not independent from its neighbors, is measured repeatedly on
        its own, sensitivity can be increased by our applied
        \textit{p}-value meta-analysis, namely Hartung's method. This
        potentially enables to call just not significant modulated
        genes based on the DGEA to be convincingly called to be part
        of a deregulated gene group.  Furthermore, it allows to
        identify connected sub-graphs, representing the propagation of
        gene regulation perturbation in the input network. A better
        understanding of this propagation, especially the critical
        spots such as sensors, effectors, and hubs, facilitates the
        projection of potential intervention points, e.g., for drug
        development. Since \texttt{MSF} only uses interaction
        information in gene regulation network, but not the functional
        grouping of the genes into functional pathways, it is
        especially adapted to discover so called cross-talk between
        such pathways.
	
	
	\section*{Conclusions}
	
	
	\texttt{MSF} is a fast and easy to use tool to find
        concertedly modulated sub-graphs in a given network. Its
        implementation in java enables its use across many operating
        systems e.g. linux and windows. So far the raw output from
        edgeR~\cite{edgeR} and DESeq2~\cite{love2014moderated} are
        supported.
	
	\subsection*{Source of data}
	
	The Ebola infection RNA-seq data set analyzed during the
        current study are available in the GEO repository
        (GSE84188)~\cite{Olejnik}. The cell signaling network file
        used is from Reactome Functional interactions (FIs) Version
        2016~\cite{Cytokegg}
	
	\subsection*{Software availability}
	
	\noindent
	Source code: \newline
	https://github.com/Modulated-Subgraph-Finder/MSF \newline Software
	Software license: MIT license.
	
	\subsection*{Supplement material}
	Supplementary material is available form GitHub: https://github.com/Modulated-Subgraph-Finder/MSF
	
	\section*{Author contributions}
	Conceptualization: FA, ILH; Funding Acquisition: ILH; Resources: ILH;
	Software: MRF; Writing – Original Draft Preparation: MRF; Writing – Review
	\& Editing: FA, ILH. All authors read and approved the final manuscript.
	
	\subsection*{Competing interests}
	
	The authors declare that they have no competing interests.
	
	\subsection*{Grant information}
	
	This work was funded by the FWF (“Fonds zur F{\"o}rderung der
        wissenschaftlichen Forschung”) within the project
        Internationalen Kooperationsprojektes - Intl cooperation
        Project (Joint Project - Lead Agency Verfahren) with the
        project number (I 1988-B22). The grant was assigned to ILH. FA
        was funded by the Austrian Science Fund (FWF) project SFB F43.
	
	
	{\small\bibliographystyle{unsrtnat}
		\bibliography{MSF_Paper_FA}}
	
	\bigskip
	
	
	
	\begin{figure*}
		\centering
		\fbox{
			\begin{minipage}{17 cm}
				\includegraphics[width=1.0\linewidth]{AlgorithmMSF.png}
				\caption{Graphical representation of the \texttt{MSF}
					heuristical approach to detect modulated sub-graphs
					in a global gene regulatory network.}
				\label{fig:pseudocodemsf}
			\end{minipage}
		}
		
	\end{figure*}
	
	\begin{figure*}
		\centering
		\fbox{
			\begin{minipage}{18 cm}
				\includegraphics[width=1.0\linewidth]{DirectedMerged_Network.png}
				\caption{Visualisation of the three modulated
					directed sub-graphs identified by \texttt{MSF} at 6~hpi. The
					node coloring is associated to KEGG pathways
					referring to the colors in the legend. The
					graph edges are from Reactome. \label{fig:Sub-graph6hpi}}
			\end{minipage}
		}
	\end{figure*}
	
	\begin{figure*}[p]
		\centering
		\fbox{
			\begin{minipage}{13 cm}
				\includegraphics[width=13cm]{MSF_Robustness.png}
				\caption{Recall rates for genes in
                                  \texttt{MSF} identified sub-graph for the
                                  three different time points of EBOV
                                  infection data for 100
                                  simulations where Poisson
                                  distributed noise was added to the
                                  experimental deduced reads per gene
                                  counts.\label{fig:DEGvsMSF}}
				\label{label}
			\end{minipage}
		}
	\end{figure*}
	
	\begin{figure*}
		\centering
		\fbox{
			\begin{minipage}{12 cm}
				\includegraphics[width=0.9\linewidth]{MSFvsJActiveModule.png}
				\caption{The Venn diagram shows the common genes identified as modulated from \texttt{MSF} identified sub-graphs and jActiveModule identified module.}
				\label{fig:MSFvsJActiveModule}
			\end{minipage}
		}
	\end{figure*}
	
	
	
	\begin{figure*}[p]
		\centering
		\fbox{
			\begin{minipage}{15 cm}
				\includegraphics[width=1.01\linewidth]{ReactomeVsMSF.png}
				\caption{The Upset plot shows the
                                  number of shared pathways between
                                  \texttt{MSF} identified sub-graph
                                  gene list and DEG cut-off list for
                                  the three time-points. All
                                  10 different
                                  toll-like receptor cascades are
                                   in the set
                                    of 164 shared
                                  pathways only between \texttt{MSF}
                                  at different time-points.}
				\label{fig:msfvsreactome}
			\end{minipage}
		}
	\end{figure*}
	
	
	
	\begin{table*}[]
		\centering
		\caption{Comparison of connected sub-graphs of
                  modulated genes in the global network identified
                  after the analysis with \texttt{MSF} and applying
                  different \textit{p}-value cut-offs from edgeR to
                  genes in MSF identified modulated sub-graphs. }
		\label{tab:rawVsHartung}
		\begin{tabular}{lll}
			\hline
			\multicolumn{1}{|l|}{}                            & \multicolumn{1}{c|}{\textbf{Total number of}}       & \multicolumn{1}{c|}{\textbf{Number of connected}}      \\
			\multicolumn{1}{|l|}{}                            & \multicolumn{1}{c|}{\textbf{genes in networks}}      & \multicolumn{1}{c|}{\textbf{sub-graph in network}}     \\ \hline
			\multicolumn{1}{|l|}{\textbf{6hpi}}               & \multicolumn{1}{l|}{}                                                & \multicolumn{1}{c|}{}                 \\ \hline
			\multicolumn{1}{|l|}{edgeR + MSF}                 & \multicolumn{1}{c|}{250}                                             & \multicolumn{1}{c|}{3}                \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.1}  & \multicolumn{1}{c|}{166}                                             & \multicolumn{1}{c|}{87}               \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.05} & \multicolumn{1}{c|}{152}                                             & \multicolumn{1}{c|}{89}               \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.01} & \multicolumn{1}{c|}{125}                                             & \multicolumn{1}{c|}{76}               \\ \hline
			\multicolumn{1}{|l|}{\textbf{24hpi}}              & \multicolumn{1}{c|}{}                                                & \multicolumn{1}{c|}{}                 \\ \hline
			\multicolumn{1}{|l|}{edgeR + MSF}                 & \multicolumn{1}{c|}{656}                                             & \multicolumn{1}{c|}{7}                \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.1}  & \multicolumn{1}{c|}{457}                                             & \multicolumn{1}{c|}{183}               \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.05} & \multicolumn{1}{c|}{418}                                             & \multicolumn{1}{c|}{198}              \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.01} & \multicolumn{1}{c|}{332}                                             & \multicolumn{1}{c|}{216}              \\ \hline
			\multicolumn{1}{|l|}{\textbf{48hpi}}              & \multicolumn{1}{c|}{}                                                & \multicolumn{1}{c|}{}                 \\ \hline
			\multicolumn{1}{|l|}{edgeR + MSF}                 & \multicolumn{1}{c|}{744}                                             & \multicolumn{1}{c|}{6}                \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.1}  & \multicolumn{1}{c|}{514}                                             & \multicolumn{1}{c|}{189}               \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.05} & \multicolumn{1}{c|}{468}                                             & \multicolumn{1}{c|}{206}               \\ \hline
			\multicolumn{1}{|l|}{p-value $\leq$ 0.01} & \multicolumn{1}{c|}{363}                                             & \multicolumn{1}{c|}{241}               \\ \hline
		\end{tabular}
	\end{table*}
	
	
	
	
\end{document}

% LocalWords:  MSF Mariam Farman Ivo Perutz microarray DGEA differentially
% LocalWords:  effectors DEGs KEGG Reactome STAT1 FCS ANOVA Kolmogorov IFN
% LocalWords:  Smirnov regulon StringApp Cytoscape transduction github FIs
% LocalWords:  vicinal Hartung Hartung's iteratively monocyte macrophages
% LocalWords:  MDMs GSE84188 EBOV Filoviridea dendritic pathogenesis edgeR
% LocalWords:  hpi cytokines chemokines CXCL10 CCL8 IL6 IL27 IL23 IFNB1
% LocalWords:  IFNA1 subgraphs INFB1 cytokine chemokine jak TNF apoptosis
% LocalWords:  ECM glycoprotein cytosolic INFA1 apoptotic Reactome's ILH
% LocalWords:  DESeq2 MRF FWF Fonds zur Forderung der wissenschaftlichen
